osrf/mbzirc
# Adds a simple demo with multiple robots lifting the same overweight block

> Note: Not sure if we really should be keeping this stuff in `mbzir_ign` or we should have a `mbzirc_demos` package.

This demo demonstrates using multiple robots to lift the same overweight block.

Requires gazebosim/gz-physics#345

## To Test (Edited by Aaron)
Make sure you have the right branch of gz-physics set up. Launch the test world file:
```bash
ign gazebo /your/path/to/mbzirc_ws/src/mbzirc/mbzirc_ign/worlds/test/multi_drone_lift.sdf
```
In another terminal run:
```bash
ros2 launch mbzirc_ign multi_uav_lift_demo.launch.py
```
You should see the drones spawn. After that run the simulators and let the robots fall onto the block. When the robots are resting on the block run:
```bash
ros2 run mbzirc_ign multi_uav_demo.py -z 3
```
You should see the robots lift the block up.
![multi-robot-coordination](https://user-images.githubusercontent.com/542272/167570421-734148e9-68ac-4d5b-88bc-daf86f01eeb5.gif)

For the dragging demo, replace the above steps with
```bash
# launch world
ign gazebo /your/path/to/mbzirc_ws/src/mbzirc/mbzirc_ign/worlds/test/multi_drone_drag.sdf

# launch demo
ros2 launch mbzirc_ign multi_uav_drag_demo.launch.py

# start drones
ros2 run mbzirc_ign multi_uav_demo.py -z 3 -x 1 -y 4

# to detach
ros2 run mbzirc_ign multi_uav_demo.py -d
```

Signed-off-by: Arjo Chakravarty <arjo@openrobotics.org>
